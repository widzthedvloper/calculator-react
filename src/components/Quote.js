import React from 'react';

const Quote = () => {
  const quote = {
    head: 'Math Magician',
    quote: 'Mathematics is not about numbers, equations, computations or algorithms: it is about understanding. â€“William Paul Thurston',
  };

  return (
    <div>
      <h1>{quote.head}</h1>
      <p>{quote.quote}</p>
    </div>
  );
};

export default Quote;
