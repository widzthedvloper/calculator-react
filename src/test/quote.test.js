import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Quote from '../components/Quote';

it('Should contain the head', () => {
  const { getByText } = render(<Quote />);
  const head = getByText(/Math Magician/);
  expect(head).toBeInTheDocument();
});
it('Should contain the quote', () => {
  const { getByText } = render(<Quote />);
  const head = getByText(/Mathematics is not about numbers, equations, computations or algorithms: it is about understanding. â€“William Paul Thurston/);
  expect(head).toBeInTheDocument();
});
