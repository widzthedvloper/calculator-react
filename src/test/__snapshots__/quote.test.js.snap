// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home link correctly 1`] = `
<h1>
  Math Magician
</h1>
`;

exports[`renders home link correctly 2`] = `
<p>
  Mathematics is not about numbers, equations, computations or algorithms: it is about understanding. â€“William Paul Thurston
</p>
`;
